generate-stackbrew-library.sh

```diff
@@ -62,55 +62,55 @@ EOH
 
 # prints "$2$1$3$1...$N"
 join() {
-	local sep="$1"; shift
-	local out; printf -v out "${sep//%/%%}%s" "$@"
-	echo "${out#$sep}"
+  local sep="$1"; shift
+  local out; printf -v out "${sep//%/%%}%s" "$@"
+  echo "${out#$sep}"
 }
 
 # Generate each of the tags.
 versions=( */ )
 versions=( "${versions[@]%/}" )
 for version in "${versions[@]}"; do
-	variants=( $version/*/ )
-	variants=( $(for variant in "${variants[@]%/}"; do
-		echo "$(basename "$variant")"
-	done) )
-	for variant in "${variants[@]}"; do
-		commit="$(dockerfileCommit "$version/$variant")"
-
-		versionAliases=( )
-		versionPostfix=""
-
-		versionAliases+=( "$version$versionPostfix" )
-		if [ "$version" == *-rc ]; then
-			versionAliases+=( "rc" )
-		fi
-		if [ "$version" == *-dev ]; then
-			versionAliases+=( "dev" )
-		fi
-
-		for channel in "${!release_channel[@]}"; do
-			if [ "$version" = "${release_channel[$channel]}" ]; then
-				versionAliases+=( "$channel" )
-			fi
-		done
-
-		variantAliases=( "${versionAliases[@]/%/-$variant}" )
-		variantAliases=( "${variantAliases[@]//latest-}" )
-
-		if [ "$variant" = "apache" ]; then
-			variantAliases+=( "${versionAliases[@]}" )
-		fi
-
-		variantParent="$(awk 'toupper($1) == "FROM" { print $2 }' "$version/$variant/Dockerfile")"
-		variantArches="${parentRepoToArches[$variantParent]}"
-
-		cat <<-EOE
-
-			Tags: $(join ', ' "${variantAliases[@]}")
-			Architectures: $(join ', ' $variantArches)
-			GitCommit: $commit
-			Directory: $version/$variant
-		EOE
-	done
-done
\ No newline at end of file
+  variants=( $version/*/ )
+  variants=( $(for variant in "${variants[@]%/}"; do
+    basename "$variant"
+  done) )
+  for variant in "${variants[@]}"; do
+    commit="$(dockerfileCommit "$version/$variant")"
+
+    versionAliases=( )
+    versionPostfix=""
+
+    versionAliases+=( "$version$versionPostfix" )
+    if [[ "$version" == *-rc ]]; then
+      versionAliases+=( "rc" )
+    fi
+    if [[ "$version" == *-dev ]]; then
+      versionAliases+=( "dev" )
+    fi
+
+    for channel in "${!release_channel[@]}"; do
+      if [ "$version" = "${release_channel[$channel]}" ]; then
+        versionAliases+=( "$channel" )
+      fi
+    done
+
+    variantAliases=( "${versionAliases[@]/%/-$variant}" )
+    variantAliases=( "${variantAliases[@]//latest-}" )
+
+    if [ "$variant" = "apache" ]; then
+      variantAliases+=( "${versionAliases[@]}" )
+    fi
+
+    variantParent="$(awk 'toupper($1) == "FROM" { print $2 }' "$version/$variant/Dockerfile")"
+    # shellcheck disable=SC2154
+    variantArches="${parentRepoToArches[$variantParent]}"
+
+    cat <-EOE
```

Comment:
This was changed from `cat <<-EOE` to `cat <-EOE` and it is mentioned in the Travis build failure.

Reply:
yes .. because `cat <<-EOE` works only with tabs, now they are replaced with spaces and so it bricked .. I'm currently searching a different way to print ;-)